### PR TITLE
backupccl: prevent OR from linking in virtual file pointing to empty key space

### PR DESCRIPTION
**Pull request by msbutler**
See individual commits.

### CODE REVIEW

**Comment by msbutler** [comment]
unrelated test flake

File: pkg/ccl/backupccl/restore_online.go [hunk 1]
```diff
@@ -194,6 +194,9 @@ func sendAddRemoteSSTWorker(
 						entry.Span,
 					)
 				}
+				if !entry.Span.Contains(file.BackupFileEntrySpan) {
+					return errors.AssertionFailedf("file span %s is not contained in restore span %s", file.BackupFileEntrySpan, entry.Span)
+				}
```
Comment (msbutler) [inline pkg/ccl/backupccl/restore_online.go:199]: one thought: we don't really clean up linked ssts if the link phase fails partway through. I feel fine with this for private preview. Better to fail loudly then silently corrupt the db.

We could gen spans before the actual link phase but i dont feel like optimizing UX private preview restore rn.

**Review by stevendanna — COMMENTED** [review]
Just some minor nits.

A bit of a bummer to have to bring out such a big hammer, but it's better than the alternative.

File: pkg/ccl/backupccl/restore_online.go (hunk 1 above)
Comment (dt) [reply pkg/ccl/backupccl/restore_online.go:199]: We believe entry.Span == restoringSubspan, right? Should we check that too just to be sure?

Comment (msbutler) [reply pkg/ccl/backupccl/restore_online.go:199]: done and done.

**Review by dt — APPROVED** [review]

**Comment by msbutler** [comment]
TFTR!

bors r=dt

**Pull request merged**